dash/test/HDF5ArrayTest.cc

```diff
@@ -457,6 +462,126 @@ TEST_F(HDF5ArrayTest, CustomType)
   });
 }
 
+/** TODO currently highly experimental */
+TEST_F(HDF5ArrayTest, AsyncIO)
+{
+  int  ext_x  = dash::size() * 1;
+#ifndef DASH_DEBUG
+  long lext_x = 1024*1024*10; // approx. 40 MB
+#else
+  long lext_x = ext_x*2;
+#endif
+  std::string mpi_impl = dash::util::Config::get<std::string>("DART_MPI_IMPL");
+  
+  if(mpi_impl == "mpich"){
+    SKIP_TEST_MSG("concurrency problems in MPICH");
+  }
+  if(dash::myid() == 0){
+    const char * thread_support = dash::is_multithreaded() ? "true" : "false";
+    LOG_MESSAGE("DART Thread support: %s", thread_support);
+  }
+    
+  double secret[] = {10, 11, 12};
+  {
+    dash::Array<double> array_a(ext_x);
+    dash::Array<double> array_b(lext_x);
+    dash::Array<double> array_c(ext_x);
+    
+    // This does not work in mpich, even with a massive amount of barriers
+    // Fill
+    fill_array(array_a, secret[0]);
+    fill_array(array_b, secret[1]);
+    fill_array(array_c, secret[2]);
+    dash::barrier();
+    // things below work as expected
+
+
+    // Currently only works if just one container is passed
+    OutputStream os(dash::launch::async, _filename);
+    os << dio::dataset("array_a")
+      << array_a
```

Comment:
I know I'm picky on OCD level here, but: please run clang-format over your code. Let's polish the format when code is added so we don't get whitespace-commits later.